src/test/java/com/feedzai/commons/sql/abstraction/engine/impl/abs/SelectForUpdateTest.java

```diff
@@ -0,0 +1,177 @@
+        properties = new Properties() {
+
+            {
+                setProperty(JDBC, config.jdbc);
+                setProperty(USERNAME, config.username);
+                setProperty(PASSWORD, config.password);
+                setProperty(ENGINE, config.engine);
+                setProperty(SCHEMA_POLICY, "create");
+            }
+        };
+
+        engine = DatabaseFactory.getConnection(properties);
+
+        // Create table
+        DbEntity testEntity = new DbEntity.Builder()
+                .name(TEST_TABLE)
+                .addColumn(PK_COL, DbColumnType.STRING)
+                .addColumn(COOL_COL, DbColumnType.STRING)
+                .pkFields(PK_COL)
+                .build();
+        engine.addEntity(testEntity);
+
+        // Populate with first item
+        engine.persist(
+                TEST_TABLE,
+                entry().set(PK_COL, PK_COL_VAL)
+                        .set(COOL_COL, INIT_COOL_COL_VAL)
+                        .build(),
+                false
+        );
+    }
+
+    @After
+    public void cleanup() throws Exception {
+        engine.dropEntity(TEST_TABLE);
+        engine.close();
+    }
+
+    @Test
+    // Cockroach doesn't respect FOR UPDATE. It will execute transactions with Serializable isolation and we need
+    // to handle errors when updating the same item concurrently on the client side
+    // (see https://www.cockroachlabs.com/docs/v19.2/postgresql-compatibility#locking-and-for-update).
+    @Category(SkipTestCockroachDB.class)
+    public void testThatSecondClientQueryReadsValueOnlyAfterFirstClientCommit() throws Exception {
+        engine.beginTransaction();
+        queryForUpdate(engine);
+
+        final CountDownLatch secondClientSelectedForUpdate = new CountDownLatch(1);
+        final AtomicBoolean secondClientFinishedWithSuccess = new AtomicBoolean(false);
+        final Thread secondClient = new Thread(() -> {
+            DatabaseEngine engine2 = null;
+            try {
+                engine2 = DatabaseFactory.getConnection(properties);
+                engine2.beginTransaction();
+                Assertions.assertThat(queryForUpdate(engine2).get(0).get(COOL_COL).toString())
+                        .as("Second client should read value updated from first client after 1st client commit.")
+                        .isEqualTo(FINAL_COOL_COL_VAL);
+                secondClientSelectedForUpdate.countDown();
+                engine2.commit();
+            } catch (final DatabaseEngineTimeoutException e) {
+                try {
+                    // As h2 has a lockTimeout = 2 seconds, we need to retry the Assert
+                    Thread.sleep(10000);
+                    Assertions.assertThat(queryForUpdate(engine2).get(0).get(COOL_COL).toString())
+                            .as("Second client should read value updated from first client after 1st client commit.")
+                            .isEqualTo(FINAL_COOL_COL_VAL);
+                    engine2.commit();
```

Comment:
Why don't you just put the code above in a while loop with a timeout for the test? instead of duplicating it?